test/helper/helper.go

```diff
@@ -74,6 +74,29 @@ func IsDeviceStateActive(deviceId string) (bool, error) {
 	return false, err
 }
 
+func IsDeviceStateInActive(deviceId string) (bool, error) {
+	var err error
+	var resp *openapiclient.Device
+	TestApiClient := TestClient()
+	predefinedTime := 200 * time.Second // Adjust this as needed
+	retryInterval := 10 * time.Second   // Adjust this as needed
+	startTime := time.Now()
+	for time.Since(startTime) < predefinedTime {
+		resp, _, err = TestApiClient.DevicesApi.FindDeviceById(context.Background(), deviceId).Execute()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error when calling `DevicesApi.FindDeviceById``: %v\n", err)
+			return false, err
+		}
+		if resp.GetState() == "inactive" {
+			return true, nil
+		}
+
+		// Sleep for the specified interval
+		time.Sleep(retryInterval)
+	}
+	return false, err
```

Comment:
Note that this is going to return a `nil` error; instead we should return an error that describes what happened.  #339 shows how we could do that

Reply:
updated